lib/metrics/peer_count.go

```diff
@@ -58,36 +70,46 @@ type PeerConnFailure struct {
 
 func (r *PeerConnFailure) Process(db *sql.DB, errs *common.MetricErrors, data *types.TelemetryRequest) error {
 	if err := json.Unmarshal(*data.TelemetryData, &r); err != nil {
-		errs.Append(data.Id, fmt.Sprintf("Error decoding peer connection failure: %v", err))
+		errs.Append(data.ID, fmt.Sprintf("Error decoding peer connection failure: %v", err))
+		return err
+	}
+
+	tx, err := db.BeginTx(context.Background(), nil)
```

Comment:
Ideally the context should be passed down. Maybe this one could be used https://github.com/status-im/telemetry/blob/master/telemetry/server.go#L39 ?